smarts/core/vehicle.py

```diff
@@ -314,17 +316,38 @@ def build_agent_vehicle(
             SceneColors.Agent.value if trainable else SceneColors.SocialAgent.value
         )
 
+        if agent_interface.vehicle_type == "car":
+            urdf_name = "vehicle"
+        elif agent_interface.vehicle_type == "bus":
+            urdf_name = "bus"
+        else:
+            raise Exception("Vehicle type does not exist!!!")
+
+        if (vehicle_filepath is None) or not os.path.exists(vehicle_filepath):
```

Comment:
As I mentioned above, if the vehicle_filepath is a dict like `{"bus": "bus.urdf", "car": "vehicle.urdf"}`, then you can locate the urdf based on the agent_interface.vehicle_type: `urdf_filepath = vehicle_filepath[agent_interface.vehicle_type]`. Likely need to rename `vehicle_filepath` to something else.